src/Goteo/Controller/InvestController.php

```diff
@@ -302,22 +302,22 @@ public function signupAction($project_id, Request $request)
      * step2: Choose payment method
      * This method will show a Form on the view that redirects to the payment gateway
      */
-    public function selectPaymentMethodAction($project_id, Request $request)
+    public function selectPaymentMethodAction(Request $request, $project_id)
     {
         $amount = $request->query->get('amount');
-        $donate_amount = $request->query->get('donate_amount');
+        $donate_amount = $request->query->getInt('donate_amount', Config::get('donate.tip_amount'));
         $email = $request->query->has('email');
         $reward = $this->validate($project_id, $request->query->get('reward'), $amount, null, 'auto');
 
         if(!($this->skip_login && $email) && !Session::isLogged()) {
-            return $this->redirect('/invest/' . $this->project->id . '/signup?' . $this->query);
+            return $this->redirect('/invest/' . $project->id . '/signup?' . $this->query);
```

Comment:
$project does not exist here yet, we should use $project_id and then we are done.